### PR TITLE
feat: validate readable/writable paths in reader and writer open()

### PR DESCRIPTION
**Pull request by msto**
## Summary

Fixes #63

`MetricReader.open` and `MetricWriter.open` now check that the provided path is readable/writable before attempting any operations on it.

## Summary by CodeRabbit

* **Improved Error Handling**
  * Reader and writer now validate file paths up front and raise clearer filesystem errors for missing paths, directory targets, broken symlinks, or permission issues.

* **New Utility**
  * Added reusable path preflight helpers to consistently detect read/write problems before opening files.

* **Tests**
  * Added comprehensive tests covering preflight behavior, symlink edge cases, and permission scenarios for both reading and writing.

* **Documentation**
  * Changelog updated to document the new validation behavior.

### CODE REVIEW

File: fgmetric/_paths.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    if not path.exists():
+        return FileNotFoundError(f"File does not exist: {path}")
+    if path.is_dir():
+        return IsADirectoryError(f"Path is a directory: {path}")
+    if not os.access(path, os.R_OK):
+        return PermissionError(f"File is not readable: {path}")
+    return None
+
+
+def path_write_error(path: Path | str) -> OSError | None:
```
Comment (ameynert) [inline fgmetric/_paths.py:40]: [suggestion](https://pullpo.io/cc?l=suggestion) 
Consider implementing the `force_overwrite` option in this PR as well.

If overwriting isn't allowed, and the path exists then there should be an error reported.

File: fgmetric/_paths.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    Args:
+        path: Filesystem path to check.
+
+    Returns:
+        None if `path` is writable; otherwise an unraised exception whose type and message
+        describe the problem (`FileNotFoundError`, `NotADirectoryError`, `IsADirectoryError`,
+        or `PermissionError`).
+    """
+    path = Path(path)
+    parent = path.absolute().parent
```
Comment (ameynert) [inline fgmetric/_paths.py]: [suggestion](https://pullpo.io/cc?l=suggestion) 
Consider using `resolve` here.

File: fgmetric/_paths.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    Args:
+        path: Filesystem path to check.
+
+    Returns:
+        None if `path` is readable; otherwise an unraised exception whose type and message
+        describe the problem (`FileNotFoundError`, `IsADirectoryError`, or `PermissionError`).
+    """
+    path = Path(path)
+    if not path.exists():
+        return FileNotFoundError(f"File does not exist: {path}")
```
Comment (ameynert) [inline fgmetric/_paths.py]: [question](https://pullpo.io/cc?l=question) 
Could you add a specific reporting for broken symlinks?

**Review by ameynert — APPROVED** [review]

File: fgmetric/_paths.py (hunk 1 above)
Comment (msto) [reply fgmetric/_paths.py:40]: I'll do that as a follow-up!

File: fgmetric/_paths.py (hunk 3 above)
Comment (msto) [reply fgmetric/_paths.py]: Done!

File: fgmetric/_paths.py (hunk 1 above)
Comment (msto) [reply fgmetric/_paths.py:40]: #69

File: fgmetric/_paths.py (hunk 2 above)
Comment (msto) [reply fgmetric/_paths.py]: Done!
